### ISSUE TITLE
Some HttT Maps are Trash

### ISSUE THREAD

**Issue by Hejnewar** [wesnoth/wesnoth#8058]
So I saw 2 players play HttT in MP (yes player creativity knows no bounds), but what I thought initally was not mainline campaign map but half baked beta add-on release, they look like absolute bs that hasnt been updated in 15 years. Maybe back then it was fine but right now their quality is trash. They provide no tactical value, no visual value, no value at all. I would really expect better treatment for a campaign that ends with scenario literally called "Battle for Wesnoth" damnit. 

North Elves
![Map-Screenshot_005](https://github.com/wesnoth/wesnoth/assets/41817546/a80a4556-c058-481f-95c4-8a951d2bc3fa)
Test of the Clans
![Map-Screenshot_006](https://github.com/wesnoth/wesnoth/assets/41817546/3a3f1939-d721-410a-8e1b-98a3fe15c0fd)
Battle for Wesnoth:
![Map-Screenshot_007](https://github.com/wesnoth/wesnoth/assets/41817546/acb00335-df28-4942-a442-05e6e54e0239)

Just look at how empty it is, there are not even ornaments there, most of the hexes look just randomly placed where they are without much thought and even if they should be mostly just plains like Test of the Clans that absolutely doesnt excuse how bad they look, especially when it comes to water and that artifical looking lake.

**Comment by Wedge009** [comment]
The request can certainly be phrased better.

Subjective, like #7479. Presumably #7106 would be a sub-set of this.

**Comment by Dalas121** [comment]
I hope you never see the horrors that are the DM mainline maps 😬

Jokes aside, I'd love to work on HttT sometime after I finish the DM rework.  IMO the whole campaign feels a little outdated, and is excessively long.

**Comment by knyghtmare** [comment]
Finally, someone noticed the substandard quality and the need for improvement.

**Comment by mesilliac** [comment]
Why wouldn't these maps be empty? Human territory in Wesnoth is largely grassland with large benefit given to hit-and-run tactics with mounted units. Isn't the whole point thematically of the latter part of the campaign that you are bringing the fight to an enemy in their home territory across open unfavourable terrain which is suited well for them and their knights + undead, and poorly for you and your forest elves, mages and mermen?

**Comment by Hejnewar** [comment]
> The request can certainly be phrased better.

Yes I totally agree but I preferred to write it when I was still emotionally involved. 

>Why wouldn't these maps be empty? Human territory in Wesnoth is largely grassland with large benefit given to hit-and-run tactics with mounted units. Isn't the whole point thematically of the latter part of the campaign that you are bringing the fight to an enemy in their home territory across open unfavourable terrain which is suited well for them and their knights + undead, and poorly for you and your forest elves, mages and mermen?

And how does that justifies them looking bad? I don't see any connection. I even said in original post that exactly that doesn't excuse how bad they look. You have ornaments you can place terrain other than plains in unimportant locations, possibly you can even get new art. Bad looks really aren't excusable.

**Comment by mesilliac** [comment]
Well because two of the complaints were it being "empty" and providing "no tactical value". I feel the emptiness of the land is important both thematically, and tactically.

**Comment by cooljeanius** [comment]
Current maps are iconic in their simplicity; they're fine as-is. Adding too many map decorations would distract from the units. 

> Finally, someone noticed the substandard quality and the need for improvement.

Yeah because you told him to file this for you; stop pretending like you weren't involved. Your standards for map quality are ridiculous and generally should be ignored.

**Comment by cooljeanius** [comment]
> Jokes aside, I'd love to work on HttT sometime after I finish the DM rework. IMO the whole campaign feels a little outdated, and is excessively long.

Being outdated is the whole point of it; it's the classic first campaign that says that it's finished, and shouldn't be touched.

**Comment by Hejnewar** [comment]
>Current maps are iconic in their simplicity; they're fine as-is. Adding too many map decorations would distract from the units.

Unless you send me a screen of a map so beautiful that it distracts from units I call BS.

>Yeah because you told him to file this for you; stop pretending like you weren't involved. Your standards for map quality are ridiculous and generally should be ignored.

This straight up is complete and utter BS because the game I was watching when I got enraged was no one else's but loonycyborg's and I can even find replay for this. It's MP after all.

>Being outdated is the whole point of it; it's the classic first campaign that says that it's finished, and shouldn't be touched.

Then it should be hidden from plainsight, I have not much regard for "tradition" and what I care about is players, if (new) players leave because camping looks bad then it should be changed and yes I have no proof, but I don't need one since I simply don't want to take any chances.

**Comment by Dalas121** [comment]
> Being outdated is the whole point of it; it's the classic first campaign that says that it's finished, and shouldn't be touched.

If we want to advertise HttT as a legacy campaign, I'm fine with that. It's definitely not the worst campaign, and has a lot of historical significance.

But AFAIK there's plans to center the new player experience around HttT and the Asheviere arc. If that ever does happen, I think HttT would need some modernization.

**Comment by cooljeanius** [comment]
> > Current maps are iconic in their simplicity; they're fine as-is. Adding too many map decorations would distract from the units.
> 
> Unless you send me a screen of a map so beautiful that it distracts from units I call BS.

See: any of the new TSG maps. Your idea of "beauty" is just litter and busyness.

> 
> > Yeah because you told him to file this for you; stop pretending like you weren't involved. Your standards for map quality are ridiculous and generally should be ignored.
> 
> This straight up is complete and utter BS because the game I was watching when I got enraged was no one else's but loonycyborg's and I can even find replay for this. It's MP after all.
> 

OK, explain your inappropriate [canvassing](https://en.wikipedia.org/wiki/Wikipedia:Canvassing) of him about this issue in the WMG Discord, then, if he didn't put you up to this:

<img width="700" alt="Screen Shot 2023-11-20 at 18 47 32" src="https://github.com/wesnoth/wesnoth/assets/1388660/9541ba8f-4522-45dc-b430-341e2ce6b96c">

> > Being outdated is the whole point of it; it's the classic first campaign that says that it's finished, and shouldn't be touched.
> 
> Then it should be hidden from plainsight, I have not much regard for "tradition"

Well then you suck as a dev and should leave the development of this game to people who are more respectful of those who came before us.

**Comment by Hejnewar** [comment]
>Well then you suck as a dev and should leave the development of this game to people who are more respectful of those who came before us.

Uuuu.... Wanting the game to progress and get new players, be generally more modernized is a sin now. I see.. Well you can always stay in the old versions instead of attacking people (not that I feel attacked, it was way too weak to be called a proper attack imo, I'm just using this as a random behavioural point).

**Comment by cooljeanius** [comment]
> Uuuu.... Wanting the game to progress and get new players, be generally more modernized is a sin now. 

The main appeal of Wesnoth is that it's retro; take that away from it and there's nothing left to attract people to it.

**Comment by cooljeanius** [comment]
Hejnewar and Knyght booting Windows XP: "0/10, not enough visual noise; Microsoft plz fix by scribbling all over it"
![Bliss_(Windows_XP)](https://github.com/wesnoth/wesnoth/assets/1388660/12f38aff-b6e6-4e8b-b003-8ad97c43d1b7)

**Comment by cooljeanius** [comment]
Hejnewar and Knyght playing Super Mario Bros: "0/10 not enough spikes and lasers; needs more sparkles and lens flares"
![World_1-1_Super_Mario_Bros](https://github.com/wesnoth/wesnoth/assets/1388660/13e7379b-7781-484e-bfd6-60bb7d061669)

**Comment by MechanicalDragon963** [comment]
I think Heinevar is right, the game needs to improve its quality, otherwise why bother developing it at all? Obviously, even with the current toolset, a better looking map representing flat terrain could be made. 

And even with the "new" maps Vesnot will still look like a retro game, I don't understand the point of this argument.

I think it's destructive to refuse to improve obviously poor quality game content because of some ridiculous "traditions". I want to play a quality made campaign, I don't want to see bad maps, no matter if they are 100 times some great and important in ancient times.

**Comment by cooljeanius** [comment]
> I think Hejnevar is right, the game needs to improve its quality, otherwise why bother developing it at all? Obviously, even with the current toolset, a better looking map representing flat terrain could be made.
> 

There are ways to improve the quality of the game that don't involve violating its signature look: QOL improvements and UI improvements and more informative WML diagnostics and portability easements and new translations and such.

> And even with the "new" maps Wesnoth will still look like a retro game, I don't understand the point of this argument.
> 
> I think it's destructive to refuse to improve obviously poor quality game content because of some ridiculous "traditions". I want to play a quality made campaign, I don't want to see bad maps, no matter if they are 100 times some great and important in ancient times.

I think it's destructive to destroy classic, time-tested maps that have survived over the ages, and try to replace them out of a desire to chase the latest shiny thing.

> > Hejnewar and Knyght playing Super Mario Bros: "0/10 not enough spikes and lasers; needs more sparkles and lens flares"
> 
> Unfortunately there's a reason that SMB hasn't used its 1985 graphics since, well, 1985. If Nintendo had kept using the same visuals over and over again because "it's classic", Mario would have died a long time ago.

Nintendo *does* keep using the same visuals whenever it ports the original one to new platforms, it only does new visuals when it makes an entirely new game. New visuals should be reserved for a "Wesnoth 2" or whatever a sequel would be called.

**Comment by Dalas121** [comment]
> Nintendo does keep using the same visuals whenever it ports the original one to new platforms, it only does new visuals when it makes an entirely new game. New visuals should be reserved for a "Wesnoth 2" or whatever a sequel would be called.

And they build those new visuals into the new Mario game that they release every year or two, dropping support for the old ones.  Meanwhile Wesnoth 1.0 has been out for 20 years and counting, with continuous major releases.

But the point is moot - nobody is proposing revamping the entire visual style, just changing the terrain on some existing maps. 

> There are ways to improve the quality of the game that don't involve violating its signature look: QOL improvements and UI improvements and more informative WML diagnostics and portability easements and new translations and such.

I actually very much agree with you on this - don't fix what isn't broken.  I do think there's oftentimes a temptation to change something because we can rather than because it's actually a problem.  Arcane resistances, for example, I honestly quite liked the way they were (sorry Hejnewar).

Unfortunately I do think there is room for improvement with HttT.  I personally think it's outdated, and based on this [random forum survey](https://forums.wesnoth.org/viewtopic.php?t=51916) HttT is barely average.  (if we assume the EI rework moved it to at least "above average").

> I think it's destructive to destroy classic, time-tested maps that have survived over the ages, and try to replace them out of a desire to chase the latest shiny thing.

I don't think anybody is proposing this, no?  I believe the original complaint was _"there are not even ornaments there, most of the hexes look just randomly placed where they are without much thought"_.  All Hejnewar is suggesting is that we change some of the lower-quality HttT maps to be more visually varied and tactically interesting, like many of the other mainline maps (including some HttT maps).

Unrelated to that, I do think HttT's gameplay needs improvements, but that doesn't affect it's "signature look" either.

**Comment by cooljeanius** [comment]
> > I think it's destructive to destroy classic, time-tested maps that have survived over the ages, and try to replace them out of a desire to chase the latest shiny thing.
> 
> I don't think anybody is proposing this, no? I believe the original complaint was _"there are not even ornaments there, most of the hexes look just randomly placed where they are without much thought"_. All Hejnewar is suggesting is that we change some of the lower-quality HttT maps to be more visually varied and tactically interesting, like many of the other mainline maps (including some HttT maps).

I think there's a temptation to over-ornament maps that needs to be resisted. Part of the reason I opened #6133 was because when I tried introducing my fiancée to the game, she couldn't play it because she couldn't make the units out from their backgrounds. To me, ornaments indicate, "there is something potentially interesting at this hex," and if there isn't, I end up feeling misled. Over-ornamentation is often distracting, it's just visual noise that detracts from the signal. Negative space is worth having to draw attention to the actual content.

**Comment by Pentarctagon** [comment]
@cooljeanius I'm not really sure what's with the vitriol here, but

> Well then you suck as a dev and should leave the development of this game to people who are more respectful of those who came before us.

is entirely uncalled for.

---

As for the actual topic...

If a map should look plain because it's literally in the middle of a wide open plains area then fine, but if there's new terrain which is consistent with that look while also making the map more interesting then intentionally choosing to not do that would be bizarre.

The fact that the North Elves map hasn't changed in ~7 years is not, by itself, a reason to forbid any changes in the future. There are a lot of things that haven't changed for a while simply because there's been nobody to change them rather than any other reason. Your most recent comment of

> I think there's a temptation to over-ornament maps that needs to be resisted. Part of the reason I opened https://github.com/wesnoth/wesnoth/issues/6133 was because when I tried introducing my fiancée to the game, she couldn't play it because she couldn't make the units out from their backgrounds. To me, ornaments indicate, "there is something potentially interesting at this hex," and if there isn't, I end up feeling misled. Over-ornamentation is often distracting, it's just visual noise that detracts from the signal. Negative space is worth having to draw attention to the actual content.

is something to consider. But if your concern is that maps are getting overly cluttered then you should've just said that in the first place.

**Comment by cooljeanius** [comment]
> @cooljeanius I'm not really sure what's with the vitriol here, but
> 
> > Well then you suck as a dev and should leave the development of this game to people who are more respectful of those who came before us.
> 
> is entirely uncalled for.
> 

I was keeping my vitriol proportionate to the vitriol with which Hejnewar was attacking the existing maps; I believe the laws of war allow allow for defense against aggression. If Hejnewar had just used the title that you just edited the title to, to start with, I wouldn't have had to be as insulting to him.

**Comment by Pentarctagon** [comment]
I... what? Laws of war? I'm really confused. You don't need to be insulting to him regardless.

**Comment by Hejnewar** [comment]
Well it was only after I said about tradition that I got this treatment so I feel like this is kinda not very true and without much connection. But then again, I'm not here to make an open-air museum. I have my direction and I want to go in it. 

That direction is making the game more popular. 

Generally being angry is quite benefitial oftentimes. Certainly better than turning passive aggressive for no reason. 

I see Wesnoth flagship campaign. 
I see it in bad state.
I don't want it to stay in bad state that can possibly make players not want to play anymore. 
I will make an angry issue out of it because I believe that threating it like that is not the right thing to do and it should be improved. 

Remember, we have retro versions of wesnoth available at all times, there is no reason not to just download older version if you want real retro experience. 

Also I don't think I have heard of any other person who had trouble differentiating units from terrain so thats kinda very unusual and probably would just benefit from contrast option not necessarily making maps with just flat.

**Comment by Hejnewar** [comment]
>Things that will make Wesnoth more popular:
>Making it available on more platforms

Where else?

>Translating it into more languages

Have you noticed that translation is a fan effort and if the game doesnt have fans it will not have translation not the other way around?

>Making it more accessible to players of all ages and abilities

Yes.

I see one thing that isn't there tho.
It is making the game better. Better looking, better playing, more interesting. 

Honestly I notice that in pretty much all fields there is some resistance for making things different than they are right now even if the end result will be objectively better than what we have right now.

**Comment by Pentarctagon** [comment]
> Fine, if Hejnewar would like to apologize and de-escalate, then I am willing to be talked down to rewording it as merely, "We have irreconcilable differences in map design philosophy," then.

Hejnewar calling the HttT maps trash may not be the most constructive way to word it, but it's not a personal insult against you and not something he needs to apologize to you for?

**Comment by cooljeanius** [comment]
> > Things that will make Wesnoth more popular:
> > Making it available on more platforms
> 
> Where else?

Consoles, mainly. A Nintendo Switch port in particular would help me introduce it to a lot of the gaming communities that I'm part of. Also the mobile ports are still out of date.

> > Translating it into more languages
> 
> Have you noticed that translation is a fan effort and if the game doesnt have fans it will not have translation not the other way around?

This is also true of all other aspects of the game. Writing, art, design, programming, and music are also all fan efforts, too. There's no reason that the methods Wesnoth applies to attract artists, musicians, coders, etc. can't also be applied to translators, too. Heck, you speak Polish, why haven't you gotten the Polish translation to 100% yet? `g.w.o` says it's still only at 84.01%. Bringing that number up would be a better use of your time than needlessly attacking the design of classic campaign maps. 

> > Making it more accessible to players of all ages and abilities
> 
> Yes.
> 
> I see one thing that isn't there tho. It is making the game better. Better looking, better playing, more interesting.

That's because none of these things are necessary, because it already looks good and plays well and is interesting.

> Honestly I notice that in pretty much all fields there is some resistance for making things different than they are right now even if the end result will be objectively better than what we have right now.

Your idea of "objectively better" is wrong.

**Comment by loonycyborg** [comment]
Honestly most players wouldn't notice/care much either way. What really can scare them off is such discord as appeared in this issue above. I understand that you're passionate about improving overall look and feel of the maps but getting openly over-emotional about it might result in situation as above. Also it isn't too respectful to original map maker who had to make do with tools that existed at that time. Mind you the final decision is in hands of actual maintainer(s) of the campaign, and they can see logic in using new engine features to update old campaigns. If people want to have it like before they can always download old version from sourceforge. It keeps all releases. All of them.

**Comment by Hejnewar** [comment]
>Bringing that number up would be a better use of your time than needlessly attacking the design of classic campaign maps.

Actually it wouldn't be, I don't know polish (yes we do have that standard here I can't say I'm someone who knows polish I just speak it) nor English well, my translation would be extremely mid at best, that being said including myself I managed to convince in total 4,5 people to do some translations, mind you they only let me do some myself after agreeing to having my part reviewed latter on. 

Some places have high quality standards. 

Also you should probably open an issue for the accessibility feature instead of just us discussing it here I think.

**Comment by Pentarctagon** [comment]
> That's because none of these things are necessary, because it already looks good and plays well and is interesting.

If Wesnoth was perfect as it is and had no room to improve, we wouldn't have a tiny fraction of the player count that Wesnoth used to have at its peak. More players means more people who might decide to contribute directly to mainline, or at least feel more motivated to make UMC because there would be more other players who would play, comment, and enjoy their work.

Games have advanced over the last 10+ years and Wesnoth often hasn't kept pace.

**Comment by AnekronCZ** [comment]
I'd at least say that finishing the very last scenario of the flagship Wesnoth campaign by spamming Knights (since the terrain is so flat you reliably get hits) isn't the most exciting finale I've ever experienced.

**Comment by Hejnewar** [comment]
>Also it isn't too respectful to original map maker

Maybe I should be, but then I looked at the maps again and I was like "that's low effort" (in this day and age), also if I'm not mistaken these maps aren't even the of maps, they were actually updated in 2008 but I might be wrong. If I were to guess that was the effect of fatigue already but still the end result wasn't something great. I should have added "with respect to time"...

I acknowledge that I could have worded it less aggressively but that's fine, if I weren't aggressive I wouldn't have done a lot of what I did. 

Like with Dunefolk, I just got angry that they were bad and with bs stats.

**Comment by cooljeanius** [comment]
> I'd at least say that finishing the very last scenario of the flagship Wesnoth campaign by spamming Knights (since the terrain is so flat you reliably get hits) isn't the most exciting finale I've ever experienced.

Spamming knights fits the flavor of the scenario, though. But I will agree that the Asheviere unit could be made a bit more powerful, though, and maybe her side could learn to reserve some reinforcements for later on, instead of just throwing all her units at you in a single wave...

**Comment by Tezereth** [comment]
I do think there's a lot of right and wrong in both sides of the argument.

I do agree some peoples seems to be chasing the latest new shiny thing when making reworks, and with sometimes unreasonable quality standards that do more harm than good.

But it is also true that some maps could use an update, mostly in the design section. I remember last I played the Test of the Clan scenario, it was quite a pain due to being unable to really use well the terrain to counter effectively the deadly cavalryman's threat, AND beat the bonus objective (that said, been a while since last played)
As for the last scenario, it's a massive battle in the flatlands around the capital and within the city, a battle that asks the participation of all your veterans available, I do think it's fine as is. One improvement I would make tho would be regarding Asheviere defensive capabilities (not as a unit, but rather in the map, like walls with stationary units blocking cavalries breakthrough).

When Wesnoth has quite a lot of needed improvement mostly regarding UI and QOL updates, changing things that are visual and already look fine, for no real reasons, is to me the least important.

PS : Calling things trash or outdated either with no real arguments or ones that are straight up just subjective and mocking peoples behind their backs doesn't make your point look more objective.

**Comment by Hejnewar** [comment]
Art is subjective. 

I feel like this is for me, when did I mock people behind their back? Even if I think I'm open enough.

**Comment by Hejnewar** [comment]
Even with everything siad here I believe that with time we won't be able to avoid improving the campaign. As new campaigns are being added and old ones improved the quality of HttT will only be going down in comparison. 

I honestly don't think that argument about map clutter is good because I simply never seen it used anywhere else despite map changes, new scenarios and campaign reworks changing maps so I would say that people who take care of them are competent enough and and know what they are doing when adding new things. To add to that there are some very cluttered maps in HttT as well I guess they should also be changed to be less cluttered (like S1 for me for example).

**Comment by knyghtmare** [comment]
Currently, HttT stands somewhere around the middle in terms of Quality. While in the current year, it's okay. However, yes. We cannot avoid improving it forever. There will come a time when the other lower quality campaigns have gotten their due improvements (DM, LoW, NR...EI already got it) and then HttT will then be last in the list of quality. Then it should get a refinement. 
If HttT was just a part of the mainline campaigns which it is, but it also more. **Heir to the Throne** is the **flagship** campaign. The campaign which we introduce players to the actual game. If it's just "okay", then what would players think? Rest of the game is like "just okay"? Do we want that? Do we? 

HttT currently is fine as is, in most cases, except for some scenarios:
- S19A Snow Plains, this scenario has no substance except being the "defeat enemy leader", and that flaming sword (which is hidden in the most noticeable way possible). Nothing else beyond that. We have new arctic fauna units, and arctic terrain. Why not use them in here to showcase how in-depth map design and go? We have an issue opened here already for S19A.

- S18: apart from the hidden event (of the path 3), and defeat enemy leaders, there's nothing here at all. Just one big empty map. This level's map is okay, but just needs some eventWML to make it more alive.

- S20B: Before [the community effort in 2011-2012](https://forums.wesnoth.org/viewtopic.php?t=32861), this scenario was left unfinished and unattended. This goes to show that there are some places in HttT still existing where work was left undone.

Thus, if there is supposed to be improvement, these have to be done first. However, which and what improvements to do address first must come from the SP content dev since they get the final say here.

Lastly, about the argument of preserving the classics, I believe we can offer up the legacy AOI, DM, THOT, and EI as addons so both new and returning players can get the chance to play them again. Surely, such a low effort task can be done?

**Comment by soliton-** [comment]
An issue to trash talk HttT maps devoid of any constructive points is not useful other than to create drama and waste energy which could be better spent on improving wesnoth.

IMO this should just be closed. It's barely more useful than a "make the game better" issue. Further discussion on what specifically to improve can be done on the forum.

Once there is a clear goal then opening issues here might be useful.

**Comment by Mirion147** [comment]
At the end of the day its important to realize what we're looking at here. Hjenewar noticed some maps were what he considered poor quality, Hjenewar is not a sugar coat it kinda guy, he's very opinionated and will openly speak his mind whenever he gets the chance. I know this well, I've butted heads with him more than I've seen anyone else here do. But that's who he is, that's how he approaches things he considers a problem, this isn't news. Another thing he does is speak about things which other people are passionate about and put those things down, if he considers them to be bad, or wrong, and sometimes even just for fun it feels like at times, he'll tug at your heart strings about one topic or another. Again, this is not new, I've been interacting with him for a few years now and it's always been like that, and im fairly certain other people have been here longer and interacted with him longer to know all this. None of this means he's wrong though. It's been a generally understood thought that the mainline campaigns are running behind what they should be in terms of quality, not only in maps but that's another topic. 

But what does all this mean? Do we drop everything and fix these maps because shiny is better and we have to fix it now now now? No. This is merely a calling card. An identifier, a call for action. These maps do need attention, there are tools and practices that are available now that were not when these maps were made. Wesnoth is a growing and constantly updating game, it would make sense that the flagship of the game would be the forerunner of such efforts imho. So this just brings up the discussion, to be taken very seriously given the gravity of the topic that is being discussed. And a plan should be made on how best to move forward, but thoughtfully, with care, ensuring the essence of the original feel was preserved while bringing a fresh look to it all. 

I also think that this should bring into the conversation the idea of a Wesnoth Classics Bundle where the more original state of the campaigns that are being modernized are available with the label and attention of the dev team to simply be maintained.

**Comment by Tsuki-no-Namida** [comment]
> So I saw 2 players play HttT in MP (yes player creativity knows no bounds)

I'm the other player that was present, playing with loonycyborg (even Hejnewar can confirm it: since should remember my example about pretty dolls, made when loonycyborg already left the game... my username in game is different btw),

**I would appreciate if you spend a few minutes in reading what i've to say, since many of you call that are doing this to get new players, and I'm one of the players that started loving the game, I also continuously try to bring new players.**

Years ago I was playing another game with loonycyborg, then he suggested me "try Wesnoth",
I did it, started playing campaigns (still 1.14), then found that some scenarios of these were annoying,
I then found that the savegame is simple plain text, and at that point I found what is one of the most important features of Wesnoth:
customization.
....Yes, you could say I cheated by editing savegames, but isn't the main target for players to have fun? I had a lot of it in this way.
From that point on, I completed all campaigns (even multiplayer Legends of Wesmere), half of which in 1.16 since update came in the way.
Never played add-on campaigns.

After that, I've started creating my own lame "for fun" multiplayer campaign by just editing savegames (yes, I'm too lazy to program seriously a campaign to play it just one time) and since them, in more than 60 scenarios I made, I implemented everything that came up to my mind, from customization of units/recruits/AMLAS, to even a _moveto_ event that makes peasants gather gold only on crops hexes at afternoon and if no other units are adjacent, to simulate them doing their harvest (the last of dozens of my homemade mods).

Me and loonycyborg were playing Httt (and it was me that modded it in the same lame way through editing 1st turn save file to be multiplayer) "just for a change" / "take a break" from the campaign for which I'm like a DM in an TTRPG, and we always allowed observers to recover each other in case of unexpected crashes or disconnections.
**I didn't expected that this would have ended in this discussion here above**, and here below there's my point of view:
**You all should be proud of developing or contributing to this awesome game.**
Most of the things you said are correct in at least one way,
what you need more, from my "player's point of view" is to make players aware of every feature of the game without making them have an headache...
1) first of all, make them aware that Wesnoth can be easily customized and edited as I do even if I'm a lazy noob while doing this, **I showed off this feature to my friends and they've been amused and remained in awe by that**,
so, more players coming from friends of who liked the game, and this effect can easily be recursive and reiterate itself for more players that will join with a full awareness of what this game really is.
2) Classic look vs New and cool design:
Which to choose?
That's your choice, but make it in a way that new players can easily retrieve what they want, either classic version or restyled one.
...I know, "go to add-on server, dude!".... but for beginners and lazy ppl like me is a mess to retrieve there what we want, so many of them won't do it and just leave Wesnoth.
3) Httt is ok? No, but a loyal and not dumb/lame player will find fun in playing it if he's self-aware:
I played Httt 3 times even if for me there's no comparison with Descent campaign, this last Httt run in that multiplayer way... Httt in multiplayer has been really fun so take it into account if you'll decide to update this or any other campaign.

Imho, Wesnoth is the ultimate turn-based hexagonal grid game of all times because you worked on it for 20 years, and this can be really easily seen by the amount of features in it,
and, if for a player that's not enough, he can just modify it as he wants, to make it be the game he wants and give it infinite replayability.

Sorry for the long speech and for these parts that were off-topic, but none of them was really off-topic if you're doing this discussion to attract new players, as it seemed to me.
So,
thanks for having said I've been creative in that 1st message,
and thanks to all of you for your work!

**Comment by mesilliac** [comment]
It's all well and good wanting to improve things, but if you don't pay attention to why something was good in the first place (this thing being the classic campaign that served as the backbone for the consistently ranked #1 open source game of the past two decades), but just randomly call it "trash", then "improving" it is far more likely to make it worse than to make it better.

There are a lot of comments here turning this into a fight between "progress" and "traditionalism". It isn't. If there are constructive suggestions for improvement of the campaign, they will be welcomed by everyone. The discussion here is not constructive, but destructive.

**Comment by CelticMinstrel** [comment]
Those maps don't seem all that bad to me, to be honest. It's true that they're not the _most_ interesting maps, and they could probably use at least a few non-mechanical decorations (flowers, fields, lily pads, whatever makes sense in context), but they more or less fit the theme of their respective scenarios.

> Remember, we have retro versions of wesnoth available at all times, there is no reason not to just download older version if you want real retro experience.

Aren't you forgetting that older versions probably won't run on newer machines?

> To me, ornaments indicate, "there is something potentially interesting at this hex," and if there isn't, I end up feeling misled. Over-ornamentation is often distracting, it's just visual noise that detracts from the signal. Negative space is worth having to draw attention to the actual content.

I don't think this is a good argument for not placing ornaments. Even older games like the Exile trilogy had non-functional ornaments randomly scattered around for flavour. If there's an accessibility problem, the solution isn't to not do it but rather to add an option to hide non-functional ornaments.

> An option to hide the display of overlays globally (although that might be dangerous, as in some cases they're actually necessary)

As stated, this is a terrible idea. The game would need to be able to differentiate between the ones that mean something and those that don't. This is actually quite hard to solve when you realize that some map makers may use ornaments as signals rather than pure decoration.

> An option to display map tiles as just a single color (determined by averaging all of the existing color-values in the tile's image into just a single new color-value, to be done dynamically so that it'd work with any other newly-added tiles automatically)

I don't think averaging the colour would produce nice results. Wouldn't it be better to just use the colour from the base archetype terrain? Some weird terrains that mix archetypes could also specify their own colours.

**Comment by cooljeanius** [comment]
> > An option to hide the display of overlays globally (although that might be dangerous, as in some cases they're actually necessary)
> 
> As stated, this is a terrible idea. The game would need to be able to differentiate between the ones that mean something and those that don't. This is actually quite hard to solve when you realize that some map makers may use ornaments as signals rather than pure decoration.

Yeah, after giving it some more thought, I agree that it would be too difficult, and withdraw the idea.

> > An option to display map tiles as just a single color (determined by averaging all of the existing color-values in the tile's image into just a single new color-value, to be done dynamically so that it'd work with any other newly-added tiles automatically)
> 
> I don't think averaging the colour would produce nice results. Wouldn't it be better to just use the colour from the base archetype terrain? Some weird terrains that mix archetypes could also specify their own colours.

I mean, I guess? Are you talking about reusing the code from drawing the minimap versions of the tiles?

**Comment by Hejnewar** [comment]
>Httt is ok? No, but a loyal and not dumb/lame player will find fun in playing it if he's self-aware:
I played Httt 3 times even if for me there's no comparison with Descent campaign, this last Httt run in that multiplayer way... Httt in multiplayer has been really fun so take it into account if you'll decide to update this or any other campaign.

Anything played with friend in mp is more fun than sp (at lest for me). ;) But yeah new player is not gonna be loyal to the game. 

>consistently ranked number 1 open source game

I dont see the point, the game is in bad state compared to when I started. Maybe I just want to go back to those days, or rather make it get to at least the point where it was in the past.

>campaign that served as the backbone for the consistently ranked number 1 open source game of the past two decades), but just randomly call it "trash", then "improving" it is far more likely to make it worse than to make it better.

I dont call the campaign trash I call that maps and sometimes gameplay of it. Thats big difference. Also I didnt do any "improving" this post is literally call for action for someone else, which I believed worked quiet well. Also worth mentioning again is that this campaign has literally unfinished scenarios....

>There are a lot of comments here turning this into a fight between "progress" and "traditionalism". It isn't. If there are constructive suggestions for improvement of the campaign, they will be welcomed by everyone. The discussion here is not constructive, but destructive.

IMO this is very important because it shines light on some of the things and issues that are rarely seen. In fact I believe every dev has to be progressive in some way otherwise they wouldnt be devs because they wouldnt be doing anything. I might be extremally progressive idk. In the end what kills the game is lack of updates, as long as there are updates the game will still have fans who will be waiting for something good, or new players that come with the updates.

**Comment by CelticMinstrel** [comment]
> Are you talking about reusing the code from drawing the minimap versions of the tiles?

It seems reasonable to reuse that to some extent.

> Yeah, after giving it some more thought, I agree that it would be too difficult, and withdraw the idea.

It's not impossible for sure, but a lot harder than it sounds

**Comment by MechanicalDragon963** [comment]
Okay, instead of swearing, I decided to try to improve some of Heinevar's designated maps myself. I attach screenshots and .map files.
What do you guys think about it? If the maps interest you, I'm willing to improve them based on your feedback. 
![4dc88a5b73eb7ea6](https://github.com/wesnoth/wesnoth/assets/99344973/634254a8-4720-40b3-b719-244c78159a38)
![6e96d4b79010c376](https://github.com/wesnoth/wesnoth/assets/99344973/adc02782-0e6c-44d8-89df-1f392539baf1)

**Comment by cooljeanius** [comment]
> Okay, instead of swearing, I decided to try to improve some of Heinevar's designated maps myself. I attach screenshots. What do you guys think about it? If the maps interest you, I'm willing to improve them based on your feedback. ![4dc88a5b73eb7ea6](https://user-images.githubusercontent.com/99344973/285380131-634254a8-4720-40b3-b719-244c78159a38.png) ![6e96d4b79010c376](https://user-images.githubusercontent.com/99344973/285380210-adc02782-0e6c-44d8-89df-1f392539baf1.png)

Those look worse; in the first one, the area around Weldyn is no longer the wide-open plains that they're supposed to be, and the additional trees make it confusable with the Ka'lian. The second one is now no longer recognizable as the "North Elves" map anymore, the new ugly brown color just ruins it. Please, stop making them busier, and just put them back to their original simplicity.

**Comment by MechanicalDragon963** [comment]
I've made some minor changes. Removed the extra forest from Veldin and made the colors of the North Elf grass closer to the original. Now my versions are closer to the original concept.
![wesnoth](https://github.com/wesnoth/wesnoth/assets/99344973/fad89465-50f5-4905-9af7-6ab5a054a69a)
![elves](https://github.com/wesnoth/wesnoth/assets/99344973/d14a29ca-615d-4e04-8585-3b59085ea159)

**Comment by knyghtmare** [comment]
> I've made some minor changes. Removed the extra forest from Veldin and made the colors of the North Elf grass closer to the original. Now my versions are closer to the original concept.

I like the improvements made here. Really looks better than the bland ones currently.

> Those look worse; in the first one, the area around Weldyn is no longer the wide-open plains that they're supposed to be, and the additional trees make it confusable with the Ka'lian. The second one is now no longer recognizable as the "North Elves" map anymore, the new ugly brown color just ruins it. Please, stop making them busier, and just put them back to their original simplicity.

Your comments seem borderline flaming at this point. I get it, you don't want change. 
However, that does not mean you can trample over someone's hours in map making by saying it's worse.
This is not how open source contributing works. We do not discourage contributions like this.
Also, Where was this passionate defending when AOI was kicked?

Maybe it's not now, but 4-5 years later when the other mainline campaigns have ascended in quality and HttT finds itself on the lower end of the spectrum, then maybe then there will be the "dire" need of improvement. 
Also, why is HttT being pushed for updates now, when we have other campaigns which need more attention for updates?

Lastly, I would really wish to read what the SP content dev has to say on all of this? There seems to be no organized effort to funnel effort for improvement right now.

**Comment by AnekronCZ** [comment]
I like the brown version better, as it looks more northern and the villages look less like they were just procedurally generated in a grid on top of the original map. Literally any of these are preferable to the originals by me. Weldyn in particular finally looks like a city, not like a small fortified settlement. Also the maps look much more natural and less geometrical. Adding deep water is also a nice touch + I appreciate the added dock.

**Comment by Hejnewar** [comment]
>Also, why is HttT being pushed for updates now, when we have other campaigns which need more attention for updates?

That's very simple, I saw something bad and decided that it is bad enough to act upon it. I very rarely play sp campaigns, that's very apparent I think given that I saw this on MP server. 

You can say that other campaigns need improvement first but I didn't see it myself and no one else got passionate enough about it to act. That's the whole reason.

**Comment by cooljeanius** [comment]
> > I've made some minor changes. Removed the extra forest from Veldin and made the colors of the North Elf grass closer to the original. Now my versions are closer to the original concept.
> 
> I like the improvements made here. Really looks better than the bland ones currently.
> 
> > Those look worse; in the first one, the area around Weldyn is no longer the wide-open plains that they're supposed to be, and the additional trees make it confusable with the Ka'lian. The second one is now no longer recognizable as the "North Elves" map anymore, the new ugly brown color just ruins it. Please, stop making them busier, and just put them back to their original simplicity.
> 
> Your comments seem borderline flaming at this point. I get it, you don't want change. However, that does not mean you can trample over someone's hours in map making by saying it's worse. This is not how open source contributing works. We do not discourage contributions like this.

Open source projects absolutely discourage destructive "contributions" when they make things worse; most of the editing that happens on Wikipedia consists of reverting other people's misguided attempts at changing things, for instance. Opposing these map changes is in the same spirit of most of the everyday reversion of Wikipedia vandalism that goes on. And, the amount of time put into it doesn't matter. Putting more time into doing something destructive doesn't stop it from being destructive. I'm saying it's worse because it *is* in fact worse. The current maps have the virtues of simplicity, comfort, recognizability, coziness, understandability, and distinctiveness of shapes. The updates have none of those.

> Also, Where was this passionate defending when AOI was kicked?

I wasn't aware of the discussion around it being removed at the time! I would've been if I'd taken part in it. If you would like to submit a PR to add AOI back to mainline, I would absolutely support it. 

> I like the brown version better, as it looks more northern

It looks like the Scouring of the Shire took place from the end of the Lord of the Rings, like my home was just set on fire.

> Also the maps look much more natural and less geometrical. 

This is bad, geometricity is useful for gameplay and easily forming mental models of the relationships between tiles.

**Comment by Tezereth** [comment]
I'm rather on the side of those who think HttT is rather well in it's current state, I don't appreciate throwing around the word "bad" at something that could be better, but far from what I truly think is bad.
HOWEVER
I also think HttT could use a few improvement. Mech is, so far, the only one that have put effort in reworking the maps as to advance the discussion. It doesn't mean they have to be accepted directly, it's a way of seeing potential improvement and discuss around it. Throwing around that it's worse just like that isn't healthy for the debate.

> Okay, instead of swearing, I decided to try to improve some of Heinevar's designated maps myself. I attach screenshots. What do you guys think about it? If the maps interest you, I'm willing to improve them based on your feedback. ![6e96d4b79010c376](https://user-images.githubusercontent.com/99344973/285380210-adc02782-0e6c-44d8-89df-1f392539baf1.png)

Now in my opinion, I do think overall, the northern elves scenario rework manage to make the map a bit more detailed while keeping a similar experience gameplay wise. I do think there's a lot of brown grass, although it makes some sense due to the northern location of the battle. Maybe the farm hex are a bit much when it comes to visual clutter, but still, not too much of a problem.

As for the Weldyn map, I think making it more like the city it's supposed to be is a good thing, although I'm usure of the way it's presented here.

**Comment by mesilliac** [comment]
The modified versions (that are more in line with the original theme) look to me like a significant improvement.

There are still some fairly large departures from the originals:
1. the Weldyn map looks like it's now at a much smaller scale. Previously it seemed a large scale with a main town, three satellite keeps, and a sprawling countryside, but here it just seems like one small town.
2. the river there is no longer easily fordable, and not at all in the south
3. the North Elves map has changed from grassland to farmland. That's a significant cultural change from a pastoral society to an agricultural one.
4. the forest there no longer has the big elf trees, and the elf citadel has cleared land around it in stead of being integrated with the forest

I don't really have much opinion on how important these are.

The new river in the Northern Elves map looks very nice.

**Comment by mesilliac** [comment]
It does seem pretty weird to me that we never see any sheep or goats in Wesnoth. Seeing as presumably that's what they're raising on all their grassland. In particular on the edge of mountainous regions liike this a human culture would usually raise sheep and goats, as the land and climate are poorly suited to agriculture.

Options are a bit limited if the only embellishment for terrain is farmland.

**Comment by MechanicalDragon963** [comment]
> Мне кажется довольно странным, что мы никогда не видим в Весноте ни овец, ни коз. Судя по всему, именно это они и выращивают на всех своих пастбищах. В частности, на окраинах подобных горных регионов человеческая культура обычно разводила овец и коз, поскольку земля и климат плохо подходят для сельского хозяйства.
> 
> Возможности немного ограничены, если единственным украшением местности являются сельскохозяйственные угодья.

To be honest, I imagined these lands as abandoned villages . I wanted to show that there used to be thriving villages here, but because of the Ashevir Troubles and orc invasions, they fell into disrepair.

I don't think it's comfortable to depict pastoralist settlements in Wesnoth. There isn't the right terrane for that, nor the animals themselves.

**Comment by MechanicalDragon963** [comment]
I found some of your thoughts interesting, and tried to implement them on the map.

- I left the farms just across the river, in the Wesnoth lands. North of the river, I tried to depict shepherd settlements. Some of them, the ones closest to the orcs, were fenced in. Also some of the settlements have cattle pens.
- I returned the elven trees to the forest and removed the path around the castle.
- I removed some of the decorations, such as snowballs, but instead added pebbles closer to the orc strongholds.
- I changed the human fort
![изображение](https://github.com/wesnoth/wesnoth/assets/99344973/feb09450-a4e6-4579-9c63-d433331ca1b2)
![изображение](https://github.com/wesnoth/wesnoth/assets/99344973/b469b43c-f14e-48d4-84e4-b9cf4246a976)
![изображение](https://github.com/wesnoth/wesnoth/assets/99344973/fa46172b-3cc2-4992-ad9f-64c6cf5a3992)
![изображение](https://github.com/wesnoth/wesnoth/assets/99344973/dc3e5c8d-23a1-42f1-af04-bc262ba37205)
![изображение](https://github.com/wesnoth/wesnoth/assets/99344973/601b4671-4620-4ab6-89ad-d3517dd0a382)

**Comment by Hejnewar** [comment]
I think that fences should be more irregular, some should be bigger some smaller and some shouldn't have them at all.

**Comment by AnekronCZ** [comment]
> Opposing these map changes is in the same spirit of most of the everyday reversion of Wikipedia vandalism that goes on. 

I don't really feel like this comparison holds any water. You could call it vandalism if it would alter the way the maps are played for the worse; these changes are for the most part cosmetic. I feel it's fair to expect maps changing to accommodate new types of terrain and embellishments as they are added. It's not an insult to the original map maker to adjust their work to reflect the current state of Wesnoth (especially when _everything else_ changes, down to unit sprites and stats).

> This is bad, geometricity is useful for gameplay and easily forming mental models of the relationships between tiles.

Aren't both roads and plains just _flat_ gameplay wise? Also modern displays are in a different aspect ratios and higher resolutions than the ones used when the maps were first designed, meaning the geometrical emptiness strikes the player much more today than it would back then.

**Comment by Discontinuum** [comment]
> The current maps have the virtues of <...> recognizability <...> 

Recognizability for whom? New players don't know the old maps, any map is new for them, they have no previous thing to recognize. Old players? But old players don't play old campaigns too often, and anyway, we should concentrate on new players to keep the game going, shouldn't we? With only occasionally returning oldtimers the game will die out

**Comment by mesilliac** [comment]
> I don't think it's comfortable to depict pastoralist settlements in Wesnoth. There isn't the right terrane for that, nor the animals themselves.

Yes, that's just what i mean - the options for map design are a bit limited by what art assets exist. I started wondering about things like adding miniature terrain animals... but i'm not sure if that would work or make sense, even if someone was willing to draw it.

> I think that fences should be more irregular, some should be bigger some smaller and some shouldn't have them at all.

I agree - is there broken fence art? That would make a lot of sense in this context.

I like the idea of changing the type and style of village as it gets closer to the orc territory. It adds more to the setting and story without changing the gameplay.

> Recognizability for whom? New players don't know the old maps, any map is new for them, they have no previous thing to recognize.

The same maps are often reused in other campaigns. IIRC this map, or a similar one, was also used in Legend of Wesmere. So it should look easily recognisable to someone who has played either campaign when they play the other. The first iteration was not really recognisable as the same, but i think the newer ones are.

**Comment by mesilliac** [comment]
IMO this version looks really good. It enhances the story by showing the spreading wasteland that's been pillaged by the orcs while not significantly changing the map or adding too much visual clutter. The village placement now makes much more sense rather than just seeming like a randomly placed grid of evenly-spaced healing points. The degraded dirt roat in the north really does a lot to tie the map together, and the fortifications on some of the villages closest to the orcs are a nice touch.

I like the tactical possibilities of the new feeder streams in the north-west, altho i'm not sure if that's an important change gameplay-wise. They also look good and give the map a more authentic geography.

**Comment by MechanicalDragon963** [comment]
> IMO this version looks really good. It enhances the story by showing the spreading wasteland that's been pillaged by the orcs while not significantly changing the map or adding too much visual clutter. The village placement now makes much more sense rather than just seeming like a randomly placed grid of evenly-spaced healing points. The degraded dirt road in the north really does a lot to tie the map together, and the fortifications on some of the villages closest to the orcs are a nice touch.
> 
> I like the tactical possibilities of the new feeder streams in the north-west, altho i'm not sure if that's an important change gameplay-wise. They also look good and give the map a more authentic geography.

I'm glad you like the map. Actually, the tributaries won't change anything in terms of gameplay - I made them a ford, which also counts as a plain and gives the same protection and passability.

**Comment by MechanicalDragon963** [comment]
[North_Elves_Rework.txt](https://github.com/wesnoth/wesnoth/files/13463281/North_Elves_Rework.txt)

Here's the map file. Just put it in the editor/maps folder and change ".txt" to ".map". Perhaps looking at the map in-game will find room for additional improvements.

**Comment by Jonathan-Kelly** [comment]
To my eyes, these map changes may be taking the cosmetic quality sideways...

Feels like there are two main issues with their direction (which also seems to be a general trend for newer mainline maps):
* Creeping [greeble](https://en.wikipedia.org/wiki/Greeble)-ification
* Too brown (especially the grass)

By "greebles" I mean lots of small random details that help make something appear to be large and complex from a distance. But the downsides of greebles are that they lack reason or purpose and when applied over a large area they can become visually exhausting to look at or comprehend.

These downsides are sometimes used "for effect" so as to give an unsettling appearance to monsters like the Borg Cube from Star Trek or the alien organism from [The Thing](https://en.wikipedia.org/wiki/The_Thing_(1982_film)). But you should carefully decide if that is really the approach you want to take for the plains outside Lintanir or the capital city of Wesnoth.

The same goes for depicting a landscape as more brown and ruined rather than lush and green.

Only do so if it really fits the story and setting. Are the lower northlands completely ruined by the orcs at this point in time? Then sure, make this region brown and muddy with the orcs' livestock (pigs or meat-horses maybe?) standing around inside fenced in wallows. Has Weldyn decayed into slums or experienced a catastrophic fire because Queen Asheviere has focused the lion's share of the kingdom's resources into the army and her cronies instead of civil projects? Then sure, make it a sprawling organic mess rather than the carefully planned out geometric capital city with straight, efficient roads reaching out in all directions like a great spiderweb.

Otherwise, I would advise taking a different approach to updating these maps. Particularly, added details should be more clustered, with low detail open space in between the clusters, and make sure each detail cluster is either very natural looking or has some clear story or purpose behind it.

**Comment by AnekronCZ** [comment]
> Has Weldyn decayed into slums or experienced a catastrophic fire because Queen Asheviere has focused the lion's share of the kingdom's resources into the army and her cronies instead of civil projects? Then sure, make it a sprawling organic mess rather than the carefully planned out geometric capital city with straight, efficient roads reaching out in all directions like a great spiderweb.

Truly geometric cities only became a thing in modern times as far as I know. In the past, settlements constantly grew and swallowed neighboring villages. They also copied terrain features, even the largest cities like the ancient Rome (built on the seven hills). By your logic, the current map just says that Ashievere demolished all buildings in and around Weldyn and perfectly leveled the ground in order for the roads to avoid making one or two small as they exit the city.

I understand the concern about "detail for the sake of detail", but a featureless map with lackluster gameplay where you just spam knights until the queen dies wasn't the grand finale I've expected from the flagship Wesnoth campaign either.

Won't comment on brown grass beyond noting that it can be just used to portray colder climate and it wouldn't cross my mind to look at it as tainted by orcs or something by default.

**Comment by Tsuki-no-Namida** [comment]
Well, I'm not as expert as you guys, but as a player who spent hundred hours in map creation before using them in my own campaign, here I read things that upset me a bit... so I preferred to spend a bit of my time here instead of programming my next scenario.
Would be too long (and even boring for you to read) if I quote and reply to every single part so I prefer putting a list:
1) Use of ford terrain type: are you sure that new players will click to check is ford and not low water? They may end thinking that they'll be slowed and lowered in defense if they pass there;
same for fences, iirc in a Secret of the Ancients scenario they even have an active effect of stopping wolves; maybe a newbie would naturally think that they stop or at least slow units, even before having played that Secret of Ancients scenario.
2) Sheep or goats is a nice idea but not for a battle scenario: if they could, they wouid run away at the sound of battle...
3) Geometrical cities/castles: different parts of what you guys say is right;
note that cities born in Roman age (and later) as fortified outposts on plain terrain are geometrical, look at Reggio Emilia in Italy in which you could still see hexagonal geometry and the city that later expanded outside that hexagon, but cities not born as fortified outposts in plain rarely are geometric.
4) Iirc in some other campaign there was a wider version of Weldyn or Elensefar (or another one, tbh I don't remember) and that one was geometrical too (correct me if I'm wrong) so is better to check which are used or redesigned wider in other campaigns before modifying them, then you can reason about if is reasonable or not to change their gemoetry.
5) Making castles/fortresses wider: for sure it would look cool but... are you sure that player, that can summon only 6 units per turn, won't be instantly wiped by the AI that could use all his gold at turn 1 and that in new map has even more villages?
6) Deep water: sincerely i never passed in the below part of last scenario except when played it in 2 players (I directly started at south of the river), but maybe you can have same cool look in alternating tropical/middle/gray low water terrain types, this without changing gameplay.
7) Never forget that httt is designed as "for beginners", so it's really hard to balance it and to fit all at once every thing you put in this discussion at the same time: each modification may have a huge impact on httt.

Again, thanks to all of you for your efforts in improving this awesome game!

**Comment by AnekronCZ** [comment]
> Geometrical cities/castles: different parts of what you guys say is right; note that cities born in Roman age (and later) as fortified outposts on plain terrain are geometrical, look at Reggio Emilia in Italy in which you could still see hexagonal geometry and the city that later expanded outside that hexagon, but cities not born as fortified outposts in plain rarely are geometric.

Oh I agree completely, I don't think Weldyn should lose it's signature shape. It's just that if it doesn't follow it perfectly (as in Mech's attempt at remaking the map) it can have a more charitable explanation than that the part of the city burned to the ground before.

There's a "more zoomed in" Weldyn in Eastern Invasion that's being remade for 1.18. In that version, the city keeps the general shape, but the streets aren't 100% straight, there are patches of farmland and bits of urban jungle (a new terrain type) scattered around and generally it feels much more organic.

The point that the two should feel similar is excellent. I think they're maybe only around 100 years apart timeline wise.

**Comment by Discontinuum** [comment]
> Use of ford terrain type: are you sure that new players will click to check is ford and not low water? They may end thinking that they'll be slowed and lowered in defense if they pass there;
same for fences, iirc in a Secret of the Ancients scenario they even have an active effect of stopping wolves; maybe a newbie would naturally think that they stop or at least slow units, even before having played that Secret of Ancients scenario.

While the concern about fences is justified, the ford terrain concern seems too much. I'm of a better opinion about Wesnoth players :) It's 20 or so scenarios passed at this point, they should've seen fords already. Even if they are not familiar at the point, let us show them, if they are never exposed, they won't have a chance to learn

**Comment by mesilliac** [comment]
I'm not sure it's a bad idea to make them actual rivers. It seems more interesting tactically. But i'm not sure what all the gameplay concerns are as it's been years since i've played the scenario.

I think the northern elves map should be submitted as a PR for further discussion. As it is, it is at very least a good starting point for improvement. The discussion here started with too much adversarialism.

**Comment by Tsuki-no-Namida** [comment]
> While the concern about fences is justified, the ford terrain concern seems too much. I'm of a better opinion about Wesnoth players :) It's 20 or so scenarios passed at this point, they should've seen fords already. Even if they are not familiar at the point, let us show fords to them, if they are never exposed, they won't have a chance to learn

Agree, I think there's already some other scenario with "ford" terrain type in httt, but in this same campaign there's a scenario namd "Ford of Abez" that hasn't ford terrain but low water and I remember well that at my 1st run I needed to redo it from the beginning to not end stuck in the low water so maybe there are other players like me that after "Ford of Abez" will have nightmares whenever they see any water to cross, especially with dwarves, so I just wanted to warn you :) 

At the end, what causes most troubles in httt in respect to other campaigns labeled as "easy" is that in httt:
- there's a huge number of scenarios in respect of other campaigns;
- you have access to lots of unit types;
- you won't know in advance what each scenario will require;

this isn't strictly related to map design, but map design has for sure a certain influence on it.

**Comment by Gothyoba** [comment]
The PR intended to fix this issue was closed due to the HttT Rework. Is this still supposed to be open?

**Comment by cooljeanius** [comment]
> The PR intended to fix this issue was closed due to the HttT Rework. Is this still supposed to be open?

I think if there's going to be a separate rework, and the Classic edition is also going to stick around separately, then the Classic maps should stay Classic, and should only get reworked in the Rework.

**Comment by soliton-** [comment]
Is this still useful especially given this is now about classic HttT?
